main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.Multimaps;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.FastTravelBetweenStopsNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mobilitydata.gtfsvalidator.util.GeospatialUtil;
+
+/**
+ * Validates: all records of `trips.txt` refer to a collection of {@code GtfsStopTime} from file
+ * `stop_times.txt` of which -when sorted by `stop_sequence`- the travel speed between each stop is
+ * not above 150 km/h
+ *
+ * <p>Time complexity: <i>O(n * p)</i>, where <i>n</i> is the number of records in
+ * <i>stop_times.txt</i> and <i>p</i> is the number of points in a trip shape.
+ *
+ * <p>Generated notice: {@link FastTravelBetweenStopsNotice}.
+ */
+@GtfsValidator
+public class TripTravelSpeedValidator extends FileValidator {
+  private static final double METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR = 3.6D;
+  @Inject GtfsTripTableContainer tripTable;
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+  @Inject GtfsStopTableContainer stopTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (Entry<String, List<GtfsStopTime>> entry :
+        Multimaps.asMap(stopTimeTable.byTripIdMap()).entrySet()) {
+      List<FastTravelBetweenStopsNotice> noticesForTrip = checkSpeedAlongTrip(entry);
+      for (FastTravelBetweenStopsNotice notice : noticesForTrip) {
+        noticeContainer.addValidationNotice(notice);
+      }
+    }
+  }
+
+  /**
+   * Calculates the travel speed between {@code GtfsStopTime}. Generates a {@code
```

Comment:
I would add "for a single trip" in the top Javadocs here just for clarity. Something like:

```
Calculates the travel speed between {@code GtfsStopTimes} for a single trip. The key for the provided map {@code Entry} is the trip_id as a string, with the value being the list of {@code GtfsStopTimes} for that trip.
```

Reply:
Thinking about this more, I think the method would be much easier to read if it just took in two parameters - `(String tripId, List<GtfsStopTime> stopTimes)` - vs. passing in the `Entry` directly.